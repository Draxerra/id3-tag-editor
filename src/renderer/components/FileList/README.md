# File List

## Functionality

The file list component displays a list of files in a table. Files in the list may be selected and removed.

### Props

`<file-list>` supports the following custom component attributes:

| attribute | type | description
| --- | --- | ---
| `headers` | IHeader[] | The headers of the file list. Should be in the form `{ text: string, value: string }`.
| `items` | object[] | The array of file objects.
| `itemKey` | String | The key to a unique ID in a file object.
| `selected` | object[] | The array of selected file objects. Can be bound to using `v-model`.

### Events

`<file-list>` emits the following events:

| event | args type | description
| --- | --- | ---
| `add-click` | n/a | Fired on add button click.
| `remove-click` | n/a | Fired on remove button click.
| `selected-updated` | object[] | The updated selected file objects. Automatically emitted when using `v-model`.

### Slots

`<file-list>` exposes the followings slots:

| slot | props | description
| --- | --- | ---
| `default` | item: object | The text to display for each table row.

### Example

```html
<file-list :headers="headers"
           :items="items"
           item-key="id"
           v-model="selected"
           @add-click="addNew"
           @remove-selected="removeSelected">
    <template slot-scope="{ item }">{{ item.name }}</template>
</file-list>
```

```ts
import Vue from 'vue';
import { Component } from "vue-property-decorator";

import FileList from '~/src/renderer/components/FileList/FileList.vue';

@Component({
    components: {
        FileList,
    },
})
export default class GenericComponent extends Vue {
    private headers = [{
        name: "Filename",
        value: "name"
    }];

    private items = [{
        id: 1,
        name: "Test File"
    }, {
        id: 2,
        name: "Test File 2"
    }];
    
    private selected = [];

    private addNew() {
        this.items.push({
            id: 3,
            name: "Test File 3",
        });
    }

    private removeSelected() {
        this.items = this.items.filter((item) =>
            !this.selected.some((selectedItem) => selectedItem.id === item.id)
        );
        this.selected = [];
    }
}
```