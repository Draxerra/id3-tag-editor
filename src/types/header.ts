export interface IHeader {
    text: string;
    value: string;
}
