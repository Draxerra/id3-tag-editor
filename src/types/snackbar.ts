export interface ISnackbar {
    message: string;
    color: "success" | "info" | "error";
}
