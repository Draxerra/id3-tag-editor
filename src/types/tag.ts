export interface ITag extends IWriteableTag {
    bitrate: number;
    channels: number;
    codec: string;
    filename: string;
    length: number;
    path: string;
    samplerate: number;
    time: string;
    track: number;
}

export interface IWriteableTag {
    album: string;
    albumartist: string;
    artist: string;
    compilation: string;
    discnumber: string;
    genre: string;
    pictures: ITagPicture[];
    title: string;
    tracknumber: string;
    year: number;
}

export interface ITagPicture {
    mimetype: string;
    picture: Uint8Array;
}
