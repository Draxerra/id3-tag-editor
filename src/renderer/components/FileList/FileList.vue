<template lang="pug">
    .FileList
        v-data-table(:headers="headers"
                    :items="items"
                    select-all
                    :value="selected"
                    @input="$emit('selected-updated', $event)"
                    class="elevation-1"
                    :pagination.sync="pagination"
                    :item-key="itemKey"
                    :rows-per-page-items="[10, 20, 50]").FileListTable
            template(slot="headers"
                     slot-scope="{ headers }")
                tr.FileListTableRow.FileListTableHeaderRow
                    th.FileListTableHeader.FileListSelectAll
                        v-checkbox(:input-value="selected.length && selected.length === items.length ? true : false"
                                   :indeterminate="selected.length && selected.length !== items.length ? true : false"
                                   primary
                                   hide-details
                                   @click.native="toggleAll").FileListSelectAllCheckbox
                    th(v-for="header in headers"
                       :key="header.text"
                       :class="['column sortable', pagination.descending ? 'desc' : 'asc', header.value === pagination.sortBy ? 'active' : '']"
                       @click="changeSort(header.value)").text-xs-left.FileListTableHeader.FileListTableHeaderContent {{ header.text }}
                        v-icon(small) arrow_upward
            template(slot="items"
                     slot-scope="props")
                tr(:active="props.selected"
                   @click="props.selected = !props.selected").FileListTableRow
                    td.FileListTableCell.FileListTableSelector
                        v-checkbox(primary
                                   :input-value="props.selected"
                                   hide-details).FileListSelectCheckbox
                    td.FileListTableCell.FileListTableContent
                        slot(:item="props.item")
        v-btn(@click="$emit('add-click')"
              color="primary").FileListAddBtn Add Files
        v-btn(v-if="selected.length"
              @click="$emit('remove-click')"
              color="error").FileListRemoveBtn Clear Files
</template>

<script lang="ts">
import Vue from "vue";
import { Component, Model, Prop } from "vue-property-decorator";

import { IHeader } from "~/src/types/header";

import VBtn from "vuetify/es5/components/VBtn";
import VCheckbox from "vuetify/es5/components/VCheckbox";
import VDataTable from "vuetify/es5/components/VDataTable";
import VIcon from "vuetify/es5/components/VIcon";

@Component({
    components: {
        VBtn,
        VCheckbox,
        VDataTable,
        VIcon,
    },
})
export default class FileList extends Vue {
    @Prop({type: Array, required: true}) private headers!: IHeader[];
    @Prop({type: Array, required: true}) private items!: object[];
    @Prop({type: String, required: true}) private itemKey!: string;
    @Model("selected-updated", {type: Array, required: true}) private selected!: object[];

    private pagination = { sortBy: "", descending: false };

    private toggleAll() {
        const updatedSelected = this.selected.length === this.items.length ?
            [] :
            this.items.slice();
        this.$emit("selected-updated", updatedSelected);
    }

    private changeSort(column) {
        if (this.pagination.sortBy === column) {
            if (this.pagination.descending) {
                this.pagination.sortBy = "";
                this.pagination.descending = false;
            } else {
                this.pagination.descending = true;
            }
        } else {
            this.pagination.sortBy = column;
            this.pagination.descending = false;
        }
    }
}
</script>

<style lang="stylus" scoped>
.FileListTable {
    /deep/ .FileListTableRow {
        cursor: pointer;
    }
}
</style>
