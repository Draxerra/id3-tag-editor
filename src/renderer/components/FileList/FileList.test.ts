import { config, mount } from "@vue/test-utils";
import FileList from "./FileList.vue";

import { IHeader } from "~/src/types/header";

import Vue from "vue";
import Vuetify from "vuetify/es5/components/Vuetify";

Vue.use(Vuetify);

const el = document.createElement("div");
el.setAttribute("data-app", "true");
document.body.appendChild(el);

config.logModifiedComponents = false;

const headers: IHeader[] = [{
    text: "Filename",
    value: "name",
}];

const items = [{
    id: 1,
    name: "Test File 1",
}, {
    id: 2,
    name: "Test File 2",
}];

describe("File List Component", () => {

    it("is a Vue instance", async () => {
        const wrapper = mount(FileList, {
            propsData: {
                headers,
                itemKey: "id",
                items,
                selected: [],
            },
            scopedSlots: {
                default: "<template slot-scope='props'>{{ props.item.name }}</template>",
            },
            sync: false,
        });
        await wrapper.vm.$nextTick();
        expect(wrapper.isVueInstance()).toBeTruthy();
    });

    it("snapshot has not changed", async () => {
        const wrapper = mount(FileList, {
            propsData: {
                headers,
                itemKey: "id",
                items,
                selected: [],
            },
            scopedSlots: {
                default: "<template slot-scope='props'>{{ props.item.name }}</template>",
            },
            sync: false,
        });
        await wrapper.vm.$nextTick();
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("snapshot with no items has not changed", async () => {
        const wrapper = mount(FileList, {
            propsData: {
                headers,
                itemKey: "id",
                items: [],
                selected: [],
            },
            scopedSlots: {
                default: "<template slot-scope='props'>{{ props.item.name }}</template>",
            },
            sync: false,
        });
        await wrapper.vm.$nextTick();
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("displays the headers", async () => {
        const wrapper = mount(FileList, {
            propsData: {
                headers,
                itemKey: "id",
                items,
                selected: [],
            },
            scopedSlots: {
                default: "<template slot-scope='props'>{{ props.item.name }}</template>",
            },
            sync: false,
        });
        await wrapper.vm.$nextTick();
        const headerElements = wrapper.findAll(".FileListTableHeader");
        expect(headerElements.length).toEqual(2);
        expect(headerElements.at(1).text()).toContain("Filename");
    });

    it("displays the items", async () => {
        const wrapper = mount(FileList, {
            propsData: {
                headers,
                itemKey: "id",
                items,
                selected: [],
            },
            scopedSlots: {
                default: "<template slot-scope='props'>{{ props.item.name }}</template>",
            },
            sync: false,
        });
        await wrapper.vm.$nextTick();
        const contentElements = wrapper.findAll(".FileListTableContent");
        expect(contentElements.length).toEqual(2);
        expect(contentElements.at(0).text()).toContain("Test File 1");
        expect(contentElements.at(1).text()).toContain("Test File 2");
    });

    it("clicking select all checkbox emits selected-updated event", async () => {
        const wrapper = mount(FileList, {
            propsData: {
                headers,
                itemKey: "id",
                items,
                selected: [],
            },
            scopedSlots: {
                default: "<template slot-scope='props'>{{ props.item.name }}</template>",
            },
            sync: false,
        });
        await wrapper.vm.$nextTick();
        wrapper.find(".FileListSelectAllCheckbox").trigger("click");
        const selected: typeof items = wrapper.emitted()["selected-updated"][0][0];
        expect(selected.length).toEqual(2);
        expect(selected[0].name).toEqual("Test File 1");
        expect(selected[1].name).toEqual("Test File 2");
    });

    it("clicking select all two times emits selected-updated event with blank array", async () => {
        const wrapper = mount(FileList, {
            propsData: {
                headers,
                itemKey: "id",
                items,
                selected: [],
            },
            scopedSlots: {
                default: "<template slot-scope='props'>{{ props.item.name }}</template>",
            },
            sync: false,
        });
        await wrapper.vm.$nextTick();
        wrapper.find(".FileListSelectAllCheckbox").trigger("click");
        const selected: typeof items = wrapper.emitted()["selected-updated"][0][0];
        wrapper.setProps({ selected });
        wrapper.find(".FileListSelectAllCheckbox").trigger("click");
        expect((wrapper.emitted()["selected-updated"][1][0] as typeof items).length).toEqual(0);
    });

    it("selected array length > 0 and < items array length sets select all checkbox to indeterminate", async () => {
        const wrapper = mount(FileList, {
            propsData: {
                headers,
                itemKey: "id",
                items,
                selected: [items[0]],
            },
            scopedSlots: {
                default: "<template slot-scope='props'>{{ props.item.name }}</template>",
            },
            sync: false,
        });
        await wrapper.vm.$nextTick();
        const selectAll = wrapper.find(".FileListSelectAllCheckbox");
        expect(selectAll.vm.$props.inputValue).toBeFalsy();
        expect(selectAll.vm.$props.indeterminate).toBeTruthy();
    });

    it("blank selected array sets select all checkbox to unchecked", async () => {
        const wrapper = mount(FileList, {
            propsData: {
                headers,
                itemKey: "id",
                items,
                selected: [],
            },
            scopedSlots: {
                default: "<template slot-scope='props'>{{ props.item.name }}</template>",
            },
            sync: false,
        });
        await wrapper.vm.$nextTick();
        const selectAll = wrapper.find(".FileListSelectAllCheckbox");
        expect(selectAll.vm.$props.inputValue).toBeFalsy();
        expect(selectAll.vm.$props.indeterminate).toBeFalsy();
    });

    it("selected array same as items array length sets select all checkbox to checked", async () => {
        const wrapper = mount(FileList, {
            propsData: {
                headers,
                itemKey: "id",
                items,
                selected: items,
            },
            scopedSlots: {
                default: "<template slot-scope='props'>{{ props.item.name }}</template>",
            },
            sync: false,
        });
        await wrapper.vm.$nextTick();
        const selectAll = wrapper.find(".FileListSelectAllCheckbox");
        expect(selectAll.vm.$props.inputValue).toBeTruthy();
        expect(selectAll.vm.$props.indeterminate).toBeFalsy();
    });

    it("changes sort to ascending on first sort button click", async () => {
        const wrapper = mount(FileList, {
            propsData: {
                headers,
                itemKey: "id",
                items,
                selected: [],
            },
            scopedSlots: {
                default: "<template slot-scope='props'>{{ props.item.name }}</template>",
            },
            sync: false,
        });
        const headerContent = wrapper.find(".FileListTableHeaderContent");
        headerContent.trigger("click");
        await wrapper.vm.$nextTick();
        expect(wrapper.vm.$data.pagination.sortBy).toEqual("name");
        expect(wrapper.vm.$data.pagination.descending).toEqual(false);
        expect(headerContent.classes()).not.toContain("desc");
        expect(headerContent.classes()).toContain("asc");
        expect(headerContent.classes()).toContain("active");
        const contentElements = wrapper.findAll(".FileListTableContent");
        expect(contentElements.length).toEqual(2);
        expect(contentElements.at(0).text()).toEqual("Test File 1");
        expect(contentElements.at(1).text()).toEqual("Test File 2");
    });

    it("changes sort to descending on second sort button click", async () => {
        const wrapper = mount(FileList, {
            propsData: {
                headers,
                itemKey: "id",
                items,
                selected: [],
            },
            scopedSlots: {
                default: "<template slot-scope='props'>{{ props.item.name }}</template>",
            },
            sync: false,
        });
        const headerContent = wrapper.find(".FileListTableHeaderContent");
        headerContent.trigger("click");
        headerContent.trigger("click");
        await wrapper.vm.$nextTick();
        expect(wrapper.vm.$data.pagination.sortBy).toEqual("name");
        expect(wrapper.vm.$data.pagination.descending).toEqual(true);
        expect(headerContent.classes()).not.toContain("asc");
        expect(headerContent.classes()).toContain("desc");
        expect(headerContent.classes()).toContain("active");
        const contentElements = wrapper.findAll(".FileListTableContent");
        expect(contentElements.length).toEqual(2);
        expect(contentElements.at(0).text()).toEqual("Test File 2");
        expect(contentElements.at(1).text()).toEqual("Test File 1");
    });

    it("changes sort to none on third sort button click", async () => {
        const wrapper = mount(FileList, {
            propsData: {
                headers,
                itemKey: "id",
                items,
                selected: [],
            },
            scopedSlots: {
                default: "<template slot-scope='props'>{{ props.item.name }}</template>",
            },
            sync: false,
        });
        const headerContent = wrapper.find(".FileListTableHeaderContent");
        headerContent.trigger("click");
        headerContent.trigger("click");
        headerContent.trigger("click");
        await wrapper.vm.$nextTick();
        expect(wrapper.vm.$data.pagination.sortBy).toEqual("");
        expect(wrapper.vm.$data.pagination.descending).toEqual(false);
        expect(headerContent.classes()).not.toContain("active");
        const contentElements = wrapper.findAll(".FileListTableContent");
        expect(contentElements.length).toEqual(2);
        expect(contentElements.at(0).text()).toEqual("Test File 1");
        expect(contentElements.at(1).text()).toEqual("Test File 2");
    });

    it("emits add-click event on add button click", async () => {
        const wrapper = mount(FileList, {
            propsData: {
                headers,
                itemKey: "id",
                items,
                selected: [],
            },
            scopedSlots: {
                default: "<template slot-scope='props'>{{ props.item.name }}</template>",
            },
            sync: false,
        });
        await wrapper.vm.$nextTick();
        wrapper.find(".FileListAddBtn").trigger("click");
        expect(wrapper.emitted()["add-click"]).toEqual([[]]);
    });

    it("remove button is hidden with blank selected array", async () => {
        const wrapper = mount(FileList, {
            propsData: {
                headers,
                itemKey: "id",
                items,
                selected: [],
            },
            scopedSlots: {
                default: "<template slot-scope='props'>{{ props.item.name }}</template>",
            },
            sync: false,
        });
        await wrapper.vm.$nextTick();
        expect(wrapper.find(".FileListRemoveBtn").exists()).toBeFalsy();
    });

    it("remove button is visible with selected array elements", async () => {
        const wrapper = mount(FileList, {
            propsData: {
                headers,
                itemKey: "id",
                items,
                selected: items,
            },
            scopedSlots: {
                default: "<template slot-scope='props'>{{ props.item.name }}</template>",
            },
            sync: false,
        });
        await wrapper.vm.$nextTick();
        expect(wrapper.find(".FileListRemoveBtn").exists()).toBeTruthy();
    });

    it("emits remove-click on remove button click", async () => {
        const wrapper = mount(FileList, {
            propsData: {
                headers,
                itemKey: "id",
                items,
                selected: items,
            },
            scopedSlots: {
                default: "<template slot-scope='props'>{{ props.item.name }}</template>",
            },
            sync: false,
        });
        await wrapper.vm.$nextTick();
        wrapper.find(".FileListRemoveBtn").trigger("click");
        expect(wrapper.emitted()["remove-click"]).toEqual([[]]);
    });

});
