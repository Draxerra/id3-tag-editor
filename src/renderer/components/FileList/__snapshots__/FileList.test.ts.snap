// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File List Component snapshot has not changed 1`] = `
<div class="FileList">
  <div class="elevation-1 FileListTable">
    <div class="v-table__overflow">
      <table class="v-datatable v-table v-datatable--select-all">
        <thead>
          <tr class="FileListTableRow FileListTableHeaderRow">
            <th class="FileListTableHeader FileListSelectAll">
              <div class="v-input FileListSelectAllCheckbox v-input--selection-controls v-input--checkbox v-input--hide-details" primary="primary">
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <div class="v-input--selection-controls__input">
                      <input aria-checked="false" role="checkbox" type="checkbox" primary="primary">
                      <div class="v-input--selection-controls__ripple"></div>
                      <i aria-hidden="true" class="v-icon material-icons">check_box_outline_blank</i>
                    </div>
                  </div>
                </div>
              </div>
            </th>
            <th class="text-xs-left FileListTableHeader FileListTableHeaderContent column sortable asc">Filename
              <i aria-hidden="true" class="v-icon material-icons" style="font-size: 16px;">arrow_upward</i>
            </th>
          </tr>
          <tr class="v-datatable__progress">
            <th colspan="2" class="column"></th>
          </tr>
        </thead>
        <tbody>
          <tr class="FileListTableRow">
            <td class="FileListTableCell FileListTableSelector">
              <div class="v-input FileListSelectCheckbox v-input--selection-controls v-input--checkbox v-input--hide-details" primary="primary">
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <div class="v-input--selection-controls__input">
                      <input aria-checked="false" role="checkbox" type="checkbox" primary="primary">
                      <div class="v-input--selection-controls__ripple"></div>
                      <i aria-hidden="true" class="v-icon material-icons">check_box_outline_blank</i>
                    </div>
                  </div>
                </div>
              </div>
            </td>
            <td class="FileListTableCell FileListTableContent">Test File 1</td>
          </tr>
          <tr class="FileListTableRow">
            <td class="FileListTableCell FileListTableSelector">
              <div class="v-input FileListSelectCheckbox v-input--selection-controls v-input--checkbox v-input--hide-details" primary="primary">
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <div class="v-input--selection-controls__input">
                      <input aria-checked="false" role="checkbox" type="checkbox" primary="primary">
                      <div class="v-input--selection-controls__ripple"></div>
                      <i aria-hidden="true" class="v-icon material-icons">check_box_outline_blank</i>
                    </div>
                  </div>
                </div>
              </div>
            </td>
            <td class="FileListTableCell FileListTableContent">Test File 2</td>
          </tr>
        </tbody>
      </table>
    </div>
    <div class="v-datatable v-table v-datatable--select-all">
      <div class="v-datatable__actions">
        <div class="v-datatable__actions__select">Rows per page:
          <div role="combobox" class="v-input v-text-field v-select v-input--hide-details v-input--is-label-active v-input--is-dirty">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-menu v-menu--inline">
                  <div class="v-menu__activator">
                    <div class="v-select__slot">
                      <div class="v-select__selections">
                        <div class="v-select__selection v-select__selection--comma">10</div>
                        <input aria-label="Rows per page:" readonly="readonly" type="text" autocomplete="on" aria-readonly="false">
                      </div>
                      <div class="v-input__append-inner">
                        <div class="v-input__icon v-input__icon--append">
                          <i aria-hidden="true" class="v-icon material-icons">arrow_drop_down</i>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="v-datatable__actions__range-controls">
          <div class="v-datatable__actions__pagination">1-2 of 2</div>
          <button type="button" class="v-btn v-btn--disabled v-btn--flat v-btn--icon" aria-label="Previous page" disabled="disabled">
            <div class="v-btn__content">
              <i aria-hidden="true" class="v-icon material-icons">chevron_left</i>
            </div>
          </button>
          <button type="button" class="v-btn v-btn--disabled v-btn--flat v-btn--icon" aria-label="Next page" disabled="disabled">
            <div class="v-btn__content">
              <i aria-hidden="true" class="v-icon material-icons">chevron_right</i>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
  <button type="button" class="FileListAddBtn v-btn primary">
    <div class="v-btn__content">Add Files</div>
  </button>
  <!---->
</div>
`;

exports[`File List Component snapshot with no items has not changed 1`] = `
<div class="FileList">
  <div class="elevation-1 FileListTable">
    <div class="v-table__overflow">
      <table class="v-datatable v-table v-datatable--select-all">
        <thead>
          <tr class="FileListTableRow FileListTableHeaderRow">
            <th class="FileListTableHeader FileListSelectAll">
              <div class="v-input FileListSelectAllCheckbox v-input--selection-controls v-input--checkbox v-input--hide-details" primary="primary">
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <div class="v-input--selection-controls__input">
                      <input aria-checked="false" role="checkbox" type="checkbox" primary="primary">
                      <div class="v-input--selection-controls__ripple"></div>
                      <i aria-hidden="true" class="v-icon material-icons">check_box_outline_blank</i>
                    </div>
                  </div>
                </div>
              </div>
            </th>
            <th class="text-xs-left FileListTableHeader FileListTableHeaderContent column sortable asc">Filename
              <i aria-hidden="true" class="v-icon material-icons" style="font-size: 16px;">arrow_upward</i>
            </th>
          </tr>
          <tr class="v-datatable__progress">
            <th colspan="2" class="column"></th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td colspan="2" class="text-xs-center">No data available</td>
          </tr>
        </tbody>
      </table>
    </div>
    <div class="v-datatable v-table v-datatable--select-all">
      <div class="v-datatable__actions">
        <div class="v-datatable__actions__select">Rows per page:
          <div role="combobox" class="v-input v-text-field v-select v-input--hide-details v-input--is-label-active v-input--is-dirty">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-menu v-menu--inline">
                  <div class="v-menu__activator">
                    <div class="v-select__slot">
                      <div class="v-select__selections">
                        <div class="v-select__selection v-select__selection--comma">10</div>
                        <input aria-label="Rows per page:" readonly="readonly" type="text" autocomplete="on" aria-readonly="false">
                      </div>
                      <div class="v-input__append-inner">
                        <div class="v-input__icon v-input__icon--append">
                          <i aria-hidden="true" class="v-icon material-icons">arrow_drop_down</i>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="v-datatable__actions__range-controls">
          <div class="v-datatable__actions__pagination">â€“</div>
          <button type="button" class="v-btn v-btn--disabled v-btn--flat v-btn--icon" aria-label="Previous page" disabled="disabled">
            <div class="v-btn__content">
              <i aria-hidden="true" class="v-icon material-icons">chevron_left</i>
            </div>
          </button>
          <button type="button" class="v-btn v-btn--disabled v-btn--flat v-btn--icon" aria-label="Next page" disabled="disabled">
            <div class="v-btn__content">
              <i aria-hidden="true" class="v-icon material-icons">chevron_right</i>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
  <button type="button" class="FileListAddBtn v-btn primary">
    <div class="v-btn__content">Add Files</div>
  </button>
  <!---->
</div>
`;
